### PR TITLE
Add definition for "neo4j"

### PR DESCRIPTION
**Pull request by cyrilschumacher**
Please fill in this template.

- [x] Prefer to make your PR against the `types-2.0` branch.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Run `tsc` without errors.
- [x] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

If changing an existing definition:
- [ ] Provide a URL to  documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.

### CODE REVIEW

File: neo4j/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,361 @@
+// Type definitions for node-neo4j 2.0.0
+// Project: https://github.com/thingdom/node-neo4j
+// Definitions by: Cyril Schumacher <https://github.com/cyrilschumacher>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="../request/index.d.ts"/>
```
Comment (andy-ms) [inline neo4j/index.d.ts]: Use `/// <reference types="request" />`.

File: neo4j/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+
+declare module "neo4j" {
```
Comment (andy-ms) [inline neo4j/index.d.ts]: You should be able to remove this wrapper and write things at the top level.

File: neo4j/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    type HttpCallback = (error: any, body: any) => void;
+    type SaveCallback = (error: any, node: any) => void;
+
+    /**
+     * Authentication information.
+     * @interface
+     */
+    export interface Authentication {
+        /**
+         * Password.
```
Comment (andy-ms) [inline neo4j/index.d.ts]: This doc comment isn't too useful -- below line already says that it's the password and it's a string.

File: neo4j/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+    type CypherCallback = (error: any, result: any) => void;
+    type DoneCallback = (error: any) => void;
+    type GetConstraintCallback = (error: any, constraints: any) => void;
+    type GetIndexesCallback = (error: any, indexes: any) => void;
+    type GetLabelCallback = (error: any, labels: any) => void;
+    type GetPropertyKeysCallback = (error: any, keys: any) => void;
+    type GetRelationshipTypesCallback = (error: any, types: any) => void;
+    type HasConstraintCallback = (error: any, exists: boolean) => void;
+    type HasIndexCallback = (error: any, exists: boolean) => void;
+    type HttpCallback = (error: any, body: any) => void;
```
Comment (andy-ms) [inline neo4j/index.d.ts]: If this type is only used once you might as well just write it in-line.

File: neo4j/neo4j-2.0.0-tests.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/// <reference path="index.d.ts"/>
```
Comment (andy-ms) [inline neo4j/neo4j-2.0.0-tests.ts]: Don't need this line; tsconfig should handle it.

File: neo4j/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+        "index.d.ts",
+        "neo4j-2.0.0-tests.ts"
```
Comment (andy-ms) [inline neo4j/tsconfig.json]: Just call it `neo4j-tests.ts`.

**Comment by andy-ms** [comment]
@cyrilschumacher Please merge from the latest `types-2.0` to fix tests.

**Comment by cyrilschumacher** [comment]
Thanks for your review, I have fixed.

**Pull request merged**